core/pkg/heartbeat/exporter/metadataprovider.go

```diff
@@ -0,0 +1,85 @@
+package exporter
+
+import (
+	"maps"
+
+	"github.com/opencost/opencost/core/pkg/clusters"
+)
+
+// HeartbeatMetadataProvider is an interface that provides metadata for heartbeat instances. It can be used to inject
+// custom metadata into a generic `Heartbeat` payload.
+type HeartbeatMetadataProvider interface {
+	// GetMetadata returns the metadata for new heartbeat instances.
+	GetMetadata() map[string]any
+}
+
+// ClusterInfoMetadataProvider is a `HeartbeatMetadataProvider` implementation that provides metadata about the cluster
+// leveraging a `ClusterInfoProvider` implementation.
+type ClusterInfoMetadataProvider struct {
+	clusterInfoProvider clusters.ClusterInfoProvider
+}
+
+// NewClusterInfoMetadataProvider creates a new `ClusterInfoMetadataProvider` instance. The `provider` parameter is used to
+// inject custom metadata, but can be set to `nil` if no metadata is needed.
+func NewClusterInfoMetadataProvider(provider clusters.ClusterInfoProvider) *ClusterInfoMetadataProvider {
+	return &ClusterInfoMetadataProvider{
+		clusterInfoProvider: provider,
+	}
+}
+
+// GetMetadata returns the metadata for new heartbeat instances. It uses the `ClusterInfoProvider` to get the cluster
+// information and injects it into the metadata map.
+func (c *ClusterInfoMetadataProvider) GetMetadata() map[string]any {
+	m := c.clusterInfoProvider.GetClusterInfo()
+	metadata := make(map[string]any, len(m))
+
+	for k, v := range m {
+		metadata[k] = v
+	}
+
+	return metadata
+}
+
+// LogLevelMetadataProvider is a `HeartbeatMetadataProvider` implementation that provides the log level.
+type LogLevelMetadataProvider struct {
```

Comment:
Can you think of a name that's more general? Today its LogLevel but in future it could be something else. Perhaps AgentInfo. Not great with names @mbolt35 any recommendation here?

Reply:
I think you got this thou!

Reply:
ok i did see the conversation after ishaan pinged me the PR, so idea is every new metadataprovider has its own struct. https://kubecost.slack.com/archives/C08ESKMCFA4/p1751041321071199. My question is invalidated then!